src/index.js

```diff
@@ -0,0 +1,220 @@
+import { exec } from 'child_process';
+import { sync as findUp } from 'find-up';
+import * as fs from 'fs-extra';
+import globCb from 'glob';
+import * as path from 'path';
+import semver from 'semver';
+import * as zkSnark from 'snarkjs';
+import { promisify } from 'util';
+
+const glob = promisify(globCb);
+
+zkSnark.bigInt.prototype.toJSON = function() {
+  return this.toString();
+};
+const circomBinary = findUp('node_modules/.bin/circom', { cwd: __dirname });
+const snarkjsBinary = findUp('node_modules/.bin/snarkjs', { cwd: __dirname });
+
+const Extension = {
+  Json: '.json',
+  Circom: '.circom',
+  VkProof: '.vk_proof',
+  VkVerifier: '.vk_verifier',
+  Solidity: '.sol'
+};
+
+class Snarks {
+  constructor(embark) {
+    this.embark = embark;
+    this.circuitsConfig = embark.pluginConfig;
+    this.compiledCircuitsPath = embark.config.dappPath('.embark', 'snarks');
+    this.registerEvents();
+  }
+
+  registerEvents() {
+    this.embark.registerActionForEvent(
+      // won't work as desired with "dev embark" (built in monorepo) until
+      // after first prerelease of embark v5
+      semver(this.embark.version).major >= 5 // || true // get rid of `|| true`
```

Comment:
```suggestion
      semver(this.embark.version).major >= 5
```

Reply:
Agreed. It's a draft PR, in part I put it up because I wanted to see the GitHub Actions thing run the stubbed-in test suite.  In other words, still more work and cleanup to do.

Reply:
I cleaned up this situation quite a bit; there's just one place now to manually toggle `this.isEmbark5` to true, and that will be removed before the PR comes out of draft.